python/dgl/data/utils.py

```diff
@@ -350,3 +361,395 @@ def __len__(self):
+    job=[]
+    for i in range(num_process):
+        sub_info = str_feats[i * batch_size : (i+1) * batch_size \
+                         if (i+1) * batch_size < num_feats else num_feats]
+        job.append(Process(target=embed_lang, args=(d, i, sub_info)))
+
+    for p in job:
+        p.start()
+
+    for p in job:
+        p.join()
+
+    for i in range(num_process):
+        if len(d[i]) > 0:
+            features.append(d[i])
+
+    features = np.concatenate(features)
+    if verbose:
+        print(features.shape)
+
+    return features
+
+def parse_word2vec_node_feature(str_feats, langs, verbose=False):
+    """ Parse a list of strings using word2vec encoding using NLP encoders in nlps
+
+    Parameters
+    ----------
+    str_feats : list of str
+        list of strings to encode
+
+    nlps : list of func
+        a list of nlp encoder functions
+
+    verbose : bool, optional
+        print out debug info
+        Default: False
+
+    Return
+    ------
+    numpy.array
+        the encoded features
+
+    Examples
+    --------
+
+    >>> inputs = ['hello', 'world']
+    >>> languages = ['en_core_web_lg', 'fr_core_news_lg']
+    >>> feats = parse_word2vec_node_feature(inputs, languages)
+
+    """
+    import spacy
+
+    nlps = []
+    for lang in langs:
+        nlp = spacy.load(lang)
+        nlps.append(nlp)
+
+    return parse_lang_feat(str_feats, nlps, verbose)
+
+def parse_category_single_feat(category_inputs, norm=None):
+    """ Parse categorical features and convert it into onehot encoding.
+
+    Each entity of category_inputs should only contain only one category.
+
+    Parameters
+    ----------
+    category_inputs : list of str
+        input categorical features
+    norm: str, optional
+        Which kind of normalization is applied to the features.
+        Supported normalization ops include:
+
+        (1) None, do nothing.
+        (2) `col`, column-based normalization. Normalize the data
+        for each column:
+
+        .. math::
+            x_{ij} = \frac{x_{ij}}{\sum_{i=0}^N{x_{ij}}}
+
+        (3) `row`, sane as None
+
+    Note
+    ----
+    sklearn.preprocessing.LabelBinarizer is used to convert
+    categorical features into a onehot encoding format.
+
+    Return
+    ------
+    numpy.array
+        The features in numpy array
+
+    Examples
+    --------
+
+    >>> inputs = ['A', 'B', 'C', 'A']
+    >>> feats = parse_category_single_feat(inputs)
+    >>> feats
+        array([[1.,0.,0.],[0.,1.,0.],[0.,0.,1.],[1.,0.,0.]])
+
+    """
+    lb = LabelBinarizer()
+    feat = lb.fit_transform(category_inputs)
+
+    # if there are only 2 catebories,
+    # fit_transform only create a array of [0, 1, ...]
+    if feat.shape[1] == 1:
+        f = np.zeros((feat.shape[0], 2))
+        f[range(f.shape[0]),feat.squeeze()] = 1.
+        feat = f
+
+    if norm == 'col':
+        return col_normalize(feat)
+    else:
+        return feat
+
+def parse_category_multi_feat(category_inputs, norm=None):
```

Comment:
Would user need the corresponding labels of the multi-hot vector (i.e. The first elements is for label 'A')?